### DISCUSSION TITLE
Tmux - seperate colors from data/structure

### DISCUSSION THREAD

**Discussion by windowsrefund** [mcchrish/zenbones.nvim#109] [Ideas]
Hello,

I'm very happy to have found this project and want to extend my gratitude for all the work done. That said, I do believe the tmux support could be greatly improved with some simple fine-tuning. Rather than providing a file to be sourced which includes both the colors and data, the provided file could simply include a list of standardized variables; each containing their respective color value.

```
# this file should be sourced early in tmux.conf
background="#191919"
foreground="#BBBBBB"
# ... and so on
```

The leaves the user to maintain the rest of their configuration as they see fit. The only thing they'd need to do is maintain their color references according to their preferences. For example:

`set -g pane-border-style fg="${foreground}"`

This actually becomes quite useful in cases where different status line configurations are being developed and auditioned since each "color compliant" configuration could just be sourced in as mentioned already.

I would have actually started working on a POC in order to showcase this method. However, I'm confused why [tmux.lua](https://github.com/rktjmp/shipwright.nvim/blob/master/lua/shipwright/transform/contrib/tmux.lua) is empty? Basically, I don't know how to proceed so I settled on dropping a note here in the hopes to inspire the idea.

**Comment by mcchrish** [comment]
@windowsrefund thanks for suggestion and I completely agree with your solution.
I think the lush tmux file is empty because it was never implemented. You can put a PR in there and let see if they agree. If not, then we can just have it here in zenbones.
